nifpga/session.py

```diff
@@ -390,8 +410,272 @@ def read(self):
+        Decimal. The largest expecting transfer would be a 64 bit fixed point
+        with the overflow status enabled (making an array of length 3).
+
+        Returns:
+            data (DataType.value): The data in the register in a python list.
+        """
+        data = super(_FxpRegister, self).read()
+        data = self._combine_array_of_u32_into_one_value(data)
+        return self._convert_from_read_value_to_decimal(data)
+
+    def _combine_array_of_u32_into_one_value(self, data):
+        combinedData = 0
+        for index in range(0, len(self)):
+            combinedData = (combinedData << 32 * index) + data[index]
+        if len(self) > 1:
+            overflow_bit = 1 if self._overflow_enabled else 0
+            total_num_bits = self._word_length + overflow_bit
+            combinedData = combinedData >> (32 * len(self) - total_num_bits)
+        return combinedData
+
+    def _convert_from_read_value_to_decimal(self, data):
+        """ This function converts read value from a RIO device and sets
+        the overflow bit if enabled and returns a Decimal representation of
+        the fixed point. """
+        if self._overflow_enabled:
+            self.overflow = self._get_overflow_value(data)
+            data = self._remove_overflow_bit(data)
+        if self._signed:
+            data = self._integer_twos_comp(data)
+        return Decimal(data * self._delta)
+
+    def _get_overflow_value(self, data):
+        """ Mask out all the data within the word length, leaving the overflow
+        bit. If the result after masking the the word portion of the fixed
+        point is none zero that indicates the data read has overflown."""
+        mask = 2**(self._word_length)
+        if data & mask > 0:
+            return True
+        return False
+
+    def _remove_overflow_bit(self, data):
+        """ This helper function masks out all bits not inside the word length,
+        ultimately returning a value of data missing the overflow bit. """
+        return data & (2**(self._word_length) - 1)
+
+    def _integer_twos_comp(self, data):
+        if (data & (1 << (self._word_length - 1))) != 0:
+            data = data - (1 << self._word_length)
+        return data
+
+    def write(self, data):
+        """ Converts the passed in argument into fixed point representation
+        and then writes it into the register.
+
+            Args:
+                data (any python datatype): The data to be written into the
+                register. This method will coerce the value to the nearest
+                possible fixed point representation.
+
+                self.overflow: Registers with overflow enabled will write the
+                value of the attribute into the register. Users should set this
+                attribute on the the register before calling write.
+        """
+        binary = self._convert_data_to_binary_fxp(data)
+        arrayData = []
+        for index in range(0, len(self)):
```

Comment:
I dislike the len(self)'s going on here.  The len of an FXP register should be 1.  Its going to return a single datapoint.  The len of an FXP array should be the number of FXP elements in the array.  

I haven't thought enough about whether or not it makes sense, but don't feel forced to inherit ArrayRegister here.